src/extensions/timeseries/TimeseriesViewNew/useTimeseriesModel.ts

```diff
@@ -1,3 +1,8 @@
+import { useRef } from "react"
```

Comment:
Here's the new file where we define all the timeseries model functionality and create the custom hook: useTimeseriesModel